sdk/servicebus/azure-servicebus/migration_guide.md

```diff
@@ -1,164 +1,444 @@
-| `queue_client.send(BatchMessage(["data 1", "data 2", ...]))`| `batch = queue_sender.create_message_batch()  batch.add_message(ServiceBusMessage("data 1"))  queue_sender.send_messages(batch)`| [Create and send a batch of messages](https://github.com/Azure/azure-sdk-for-python/tree/master/sdk/servicebus/azure-servicebus/samples/sync_samples/send_queue.py) |
+### Settling messages
 
-### Scheduling messages and cancelling scheduled messages 
+- Previously, the methods to settle messages (`complete()`, `abandon()`, `defer()` and `dead_letter()`) were on the messages themselves.
+These have been moved to the receiver in the new version, take the message as input and have the `message` suffix in their name.
+- The idea is to have the message represents just the data and not have the responsibility of any operation on the service side.
+Additionally, since a message cannot be settled if the receiver that was used to receive it is not alive, tying these operations
+to the receiver drives the message home better.
 
-| In v0.50 | Equivalent in v7 | Sample |
-|---|---|---|
-| `queue_client.get_sender().schedule(schedule_time_utc, message1, message2)` | `sb_client.get_queue_sender().schedule_messages([message1, message2], schedule_time_utc)` | [Schedule messages](https://github.com/Azure/azure-sdk-for-python/tree/master/sdk/servicebus/azure-servicebus/samples/sync_samples/schedule_messages_and_cancellation.py) |
-| `queue_client.get_sender().cancel_scheduled_messages(sequence_number1, sequence_number2)`| `sb_client.get_queue_sender().cancel_scheduled_messages([sequence_number1, sequence_number2])` | [Cancel scheduled messages](https://github.com/Azure/azure-sdk-for-python/tree/master/sdk/servicebus/azure-servicebus/samples/sync_samples/schedule_messages_and_cancellation.py)|
+In V0.50:
+```python
+
+with receiver:
+    received_message.complete()
+    # or
+    # received_message.abandon()
+    # received_message.defer()
+    # received_message.dead_letter()
+```
+
+In V7:
+```python
+with receiver:
+    receiver.complete_message(received_message)
+    # or
+    # receiver.abandon_message(received_message)
+    # receiver.defer_message(received_message)
+    # receiver.dead_letter_message(received_message)
+```
 
 ### Renewing lock on the received message
-| In v0.50 | Equivalent in v7 | Sample |
-|---|---|---|
-| `message.renew_lock()` | `receiver.renew_message_lock(message)` | [Renew lock on the message](https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/servicebus/azure-servicebus/samples/sync_samples/sample_code_servicebus.py) |
+
+- `renew_lock` method was moved from `Message` to `ServiceBusReceiver` and renamed to `renew_message_lock` for the same reason as settlement methods movement.
+
+In V0.50:
+```python
+with receiver:
+    received_message.renew_lock()
+```
+
+In V7:
+```python
+with receiver:
+    receiver.renew_message_lock(received_message)
+```
+
+### Scheduling messages and cancelling scheduled messages
```

Comment:
There is not much in this section other than renames. In which case I would either recommend not calling them out at all or having a separate section called "Renames" (or something nicer :) ) and put these there. Having a dedicated section is helpful if the method has moved or has changed in what it accepts as input or changes what it returns as output.

Other sections that fall under the same category are the ones for utc time and for receiving messages

Reply:
There is a change in the schedule/cancel schedule api that in v0.50 the method takes `argument list` while in v7 it takes `list` so I'd like to keep it.

I have removed the utc time section; as for the `receiving messages` sections since it's a champion scenario I tend to keep it as it is.